File: Data Structures & Algorithms/baseball-game/submission-2.ts
class Solution {
    /**
     * @param {string[]} operations
     * @return {number}
     */
    calPoints(operations: string[]): number {
        let scores: number[] = []
        for( const op of operations){
             switch (op){
                case "+":
                    let add:number = scores[scores.length-1] + scores[scores.length -2]
                    scores.push(add)
                    break;
                case "D":
                    let xtwo: number= scores[scores.length-1]*2
                    scores.push(xtwo)
                    break;
                case "C":
                    scores.pop()
                    break;
                default:
                    scores.push(parseInt(op))
            }
        }
         return scores.reduce((a,b) => a+b,0)
    }
}

File: Data Structures & Algorithms/validate-parentheses/submission-2.py
class Solution:
    def isValid(self, s: str) -> bool:
        closed ={ ')':'(',  '}':'{'  ,']':'[' }
        open = []
        for c in s:
            if c in closed and len(open)>0:
                if closed[c] == open[-1]:
                    open.pop()                   
                else:
                    return False
            else:
                open.append(c)
        return False if len(open)>0 else True


        